src/contracts/pods/EigenPod.sol

```diff
@@ -537,6 +550,7 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
          * *prior* to the proof provided in the `verifyWithdrawalCredentials` function.
          */
         proofIsForValidTimestamp(Endian.fromLittleEndianUint64(withdrawalProof.timestampRoot))
+        returns (VerifiedWithdrawal memory)
     {
         uint64 withdrawalHappenedTimestamp = _computeTimestampAtSlot(Endian.fromLittleEndianUint64(withdrawalProof.slotRoot));
```

Comment:
does this timestamp check still make sense here / is it even correct?
seems quite odd that we essentially have 3 different timestamps here -- the `oracleTimestamp`, `withdrawalProof.timestampRoot`, and converting the `withdrawalProof.slotRoot`  to a timestamp !

Reply:
G added timestamp root as a field in the proof but this line didn't get updated - thanks for noting it, fixing it now

Reply:
OK excellent, really glad to see the simplification and adoption of more consistency here.
Should we move the timestamp check (`proofIsForValidTimestamp(Endian.fromLittleEndianUint64(withdrawalProof.timestampRoot))`) outside of the loop, so we aren't checking it each time?

Reply:
nevermind, seems like we have to do this in the loop, am re-resolving this thread.

Reply:
Discussed on slack, this is the withdrawal timestamp not the oracle timestamp, it needs to be checked for each proof